### ISSUE TITLE
Multiple "ignore"

### ISSUE THREAD

**Issue by tensor5** [douglascrockford/JSLint#183]
It is not possible to define a function that ignores more than one parameter:

```js
function foo(ignore, ignore, a) {
    "use strict";
    return a;
}
```

```
Redefinition of 'ignore' from line 0.
```

The use case for this is when such function is required as a callback of another, and I don't need the first two or more parameters.

**Comment by douglascrockford** [comment]
Always put the callback first.

**Comment by tensor5** [comment]
I mean this kind of situation:

```js
function callback(ignore, ignore, buffer) {
    "use strict";
    // do something with buffer
}

require('fs').read(fd, buffer, offset, length, position, callback);
```

The callback in the Node's [read](https://nodejs.org/api/fs.html#fs_fs_read_fd_buffer_offset_length_position_callback) function takes three parameters, (err, bytesRead, buffer).

**Comment by douglascrockford** [comment]
Node is doing it incorrectly. File a bug report with them.

**Comment by tensor5** [comment]
What exactly is Node doing incorrectly? Is it the fact that it takes a callback with three arguments? `Array.prototype.map()` and `Array.prototype.forEach()` also take a callback with three arguments, and there are many more examples (look at Google Maps API).

Even if one day JSLint imposes a restriction on callback type signature, we only have control over the code that we write, not over what other people write. So how do we use other's APIs with JSLint?

**Comment by douglascrockford** [comment]
Necessary arguments such as callbacks should be listed first. Optional arguments should be listed last. 

map and forEach correctly take the callback in the first position.

Send you bug reports to the correct party.
